core/qthelper.cpp

```diff
@@ -1444,39 +1443,6 @@ fraction_t string_to_fraction(const char *str)
 	return fraction;
 }
 
-int getCloudURL(QString &filename)
```

Comment:
why is this mixed in with the introduction of the new git_state?

most everything else in the commit seems to be focused on that. The commit is also somewhat long, but if it really does NOTHING but that one thing (switch to git_state) then that's ok.
I would recommend to move all other changes (even if they are implicitly caused by that introduction of the state) into separate commits.